api/filter/http/fault.proto

```diff
@@ -63,21 +41,17 @@ message HTTPFault {
   string upstream_cluster = 3;
 
   // Specifies a set of headers that the filter should match on. The fault
-  // injection filter can be applied selectively to requests that match a
-  // set of headers specified in the fault filter config. The chances of
-  // actual fault injection further depend on the values of FaultAbort.percent
-  // and FaultDelay.percent parameters. The filter will check the request’s
-  // headers against all the specified headers in the filter config. A
-  // match will happen if all the headers in the config are present in the
-  // request with the same values (or based on presence if the `value` field
-  // is not in the config). TODO: allow runtime configuration on per entry
-  // basis for headers match.
-  repeated HeaderMatcher headers = 4 [(validate.rules).repeated.unique = true];;
-
-  // Faults are injected for the specified list of downstream hosts. If
-  // this setting is not set, faults are injected for all downstream
-  // nodes. Downstream node name is taken from the HTTP
-  // x-envoy-downstream-service-node header and compared against
-  // downstream_nodes list.
-  repeated string downstream_nodes = 5 [(validate.rules).repeated.unique = true];;
+  // injection filter can be applied selectively to requests that match a set of headers specified in
+  // the fault filter config. The chances of actual fault injection further depend on the value of
+  // the :ref:`percent <envoy_api_field_filter.http.FaultAbort.percent>` field. The filter will check the request's headers
+  // against all the specified headers in the filter config. A match will happen if all the headers in
+  // the config are present in the request with the same values (or based on presence if the *value*
+  // field is not in the config).
+  repeated HeaderMatcher headers = 4 [(validate.rules).repeated.unique = true];
```

Comment:
You can only do unique on scalar types. When I ran PGV against data-plane-api today (PR is on its way), I discovered this and other interesting violations.

Reply:
Yeah I thought this looked suspicious. Do you mind if I just merge and you can fix all the violations when you plumb it through?

Reply:
I knew it was too good to be true!.. Sorry about this. I will send a PR to PGV asking them to clarify.